system/BaseModel.php

```diff
@@ -899,6 +899,10 @@ public function insertBatch(?array $set = null, ?bool $escape = null, int $batch
      */
     public function update($id = null, $data = null): bool
     {
+        if (is_bool($id)) {
+            throw new InvalidArgumentException('$id should not be boolean.');
```

Comment:
probably more verbose message: `argument #1: $id should not be boolean`, so it immediatelly know which part that invalid.

Reply:
Ref for better verbose message https://3v4l.org/8JSTu

Reply:
Updated the exception message.